From f3257662597912619d9e9354a930fc366f77863c Mon Sep 17 00:00:00 2001
From: Alexandros Frantzis <alexandros.frantzis@collabora.com>
Date: Tue, 8 Oct 2024 17:25:57 +0300
Subject: [PATCH 14/14] media/gpu/v4l2: Don't advertise the HEVC Main10 profile

The v4l2 decoder doesn't support the HEVC Main10 profile. Since we are
not able to query the decoder for supported profiles, for now remove
Main10 from the profiles we return by default, to avoid violating
preconditions in the stateful decoder that cause it crash.
---
 media/gpu/v4l2/v4l2_utils.cc | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/media/gpu/v4l2/v4l2_utils.cc b/media/gpu/v4l2/v4l2_utils.cc
index e4b7b5c394..df0d8e21e6 100644
--- a/media/gpu/v4l2/v4l2_utils.cc
+++ b/media/gpu/v4l2/v4l2_utils.cc
@@ -355,8 +355,7 @@ static const std::map<v4l2_enum_type, std::vector<VideoCodecProfile>>
              H264PROFILE_HIGH,
          }},
 #if BUILDFLAG(ENABLE_HEVC_PARSER_AND_HW_DECODER)
-        {V4L2_CID_MPEG_VIDEO_HEVC_PROFILE,
-         {HEVCPROFILE_MAIN, HEVCPROFILE_MAIN10}},
+        {V4L2_CID_MPEG_VIDEO_HEVC_PROFILE, {HEVCPROFILE_MAIN}},
 #endif  // BUILDFLAG(ENABLE_HEVC_PARSER_AND_HW_DECODER)
         {V4L2_CID_MPEG_VIDEO_VP8_PROFILE, {VP8PROFILE_ANY}},
         {V4L2_CID_MPEG_VIDEO_VP9_PROFILE, {VP9PROFILE_PROFILE0}},
-- 
2.45.2

